src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs

```diff
@@ -50,8 +64,11 @@ public static IServiceCollection Configure<TOptions>(this IServiceCollection ser
                 throw new ArgumentNullException(nameof(configureOptions));
             }
 
-            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureOptions<TOptions>(configureOptions));
+            services.AddSingleton<IConfigureOptions<TOptions>>(new ConfigureNamedOptions<TOptions>(name, configureOptions));
             return services;
         }
+
+        public static IServiceCollection ConfigureAll<TOptions>(this IServiceCollection services, Action<TOptions> configureOptions) where TOptions : class
+            => services.Configure(name: null, configureOptions: configureOptions);
```

Comment:
Actually the relevant test is https://github.com/aspnet/Options/blob/e80c497e3df10e56a11e0a29ba6370de00b6512f/test/Microsoft.Extensions.Options.Test/OptionsSnapshotTest.cs#L196